internal/services/aws_reservation_service.go

```diff
@@ -112,9 +126,9 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var ami string
-	if strings.HasPrefix(reservation.ImageID, "ami-") {
+	if strings.HasPrefix(reservation.Detail.ImageID, "ami-") {
 		// Direct AMI ID was provided, no need to call image builder
-		ami = reservation.ImageID
+		ami = reservation.Detail.ImageID
```

Comment:
I'd vote for ImageId staying on the table. We might need to answer questions like: "how many instances have been provisioned from Image X?"

Reply:
Moved back!